docs/reference/search/field-caps.asciidoc

```diff
@@ -77,6 +77,10 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailab
   (Optional, Boolean) If `true`, unmapped fields that are mapped in one index but not in another are included in the response. Fields that don't have any mapping are never included.
   Defaults to `false`.
 
+`include_fields_with_no_value`::
+  (Optional, Boolean) If `false`, fields that never had a value in any shards are not included in the response. Fields that are not empty are always included. This flag does not consider deletions, if a field was non-empty and all the documents containing that field were deleted the field is returned even if the flag is `false`.
```

Comment:
++ thanks for the concise yet very clear and accurate explanation!

Reply:
I've been testing using the new API with our recent field_caps caching improvements. I think we should mention, that the behavior of deletions also applies to updates, something like this:
```suggestion
  (Optional, Boolean) If `false`, fields that never had a value in any shards are not included in the response. Fields that are not empty are always included. This flag does not consider deletions and update.  If a field was non-empty and all the documents containing that field were deleted or the field was removed by updates,  it will still be returned even if the flag is `false`.
```
What's interesting to me is, that if I update a document with a new field, this new field is also considered to be empty. I accidentally did that while testing:

![Discover_-_Elastic](https://github.com/elastic/elasticsearch/assets/463851/a918ab8c-fcce-4ebb-ac6e-f9dcfef2ceed)

[Link](https://kertal-pr-174063-include-fields-with-no-value.kbndev.co/wjq/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:'2023-12-29T17:30:08.220Z',to:'2024-01-18T15:23:18.313Z'))&_a=(columns:!(),filters:!(),index:'177eaeb9-6200-4792-9f17-906de91ffb9b',interval:auto,query:(language:kuery,query:''),sort:!()))